File: dataModuleTester/src/main/java/com/aneonex/bitcoinchecker/tester/util/MarketCurrencyPairsStore.java
package com.aneonex.bitcoinchecker.tester.util;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.aneonex.bitcoinchecker.datamodule.model.CurrencyPairsListWithDate;

public class MarketCurrencyPairsStore {

	private static SharedPreferences getSharedPreferences(Context context) {
		return context.getApplicationContext().getSharedPreferences("MARKET_CURRENCIY_PAIRS", Context.MODE_PRIVATE);
	}
	
	public final static void savePairsForMarket(Context context, String marketKey, CurrencyPairsListWithDate currencyPairsListWithDate) {
		try {
			savePairsStringForMarket(context, marketKey, new Gson().toJson(currencyPairsListWithDate));
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	private final static void savePairsStringForMarket(Context context, String marketKey, String jsonString) {
		getSharedPreferences(context).edit().putString(marketKey, jsonString).commit();
	}
	
	public final static CurrencyPairsListWithDate getPairsForMarket(Context context, String marketKey) {
		try {
			return new Gson().fromJson(getPairsStringForMarket(context, marketKey), CurrencyPairsListWithDate.class);
		} catch (Exception e){
			e.printStackTrace();
			return null;
		}
	}
	private final static String getPairsStringForMarket(Context context, String marketKey) {
		return getSharedPreferences(context).getString(marketKey, null);
	}
}

File: dataModuleTester/src/main/java/com/aneonex/bitcoinchecker/tester/util/MarketCurrencyPairsStore.kt
package com.aneonex.bitcoinchecker.tester.util

import android.content.Context
import android.content.SharedPreferences
import com.aneonex.bitcoinchecker.datamodule.model.CurrencyPairsListWithDate
import com.google.gson.Gson

object MarketCurrencyPairsStore {
    private fun getSharedPreferences(context: Context): SharedPreferences {
        return context.applicationContext.getSharedPreferences("MARKET_CURRENCY_PAIRS", Context.MODE_PRIVATE)
    }

    fun savePairsForMarket(context: Context, marketKey: String, currencyPairsListWithDate: CurrencyPairsListWithDate?) {
        try {
            savePairsStringForMarket(context, marketKey, Gson().toJson(currencyPairsListWithDate))
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun savePairsStringForMarket(context: Context, marketKey: String, jsonString: String) {
        getSharedPreferences(context).edit().putString(marketKey, jsonString).apply()
    }

    fun getPairsForMarket(context: Context, marketKey: String): CurrencyPairsListWithDate? {
        val pairsJson = getPairsStringForMarket(context, marketKey) ?: return null

        return try {
            Gson().fromJson(pairsJson, CurrencyPairsListWithDate::class.java)
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }

    private fun getPairsStringForMarket(context: Context, marketKey: String): String? {
        return getSharedPreferences(context).getString(marketKey, null)
    }
}